Fix QGIS crash on plugin initialization

visr opened the pull request:
Fixes #993, by creating the column count first before setting their properties, and changing the index to 0 (0-based, first), since we now have only one column.

Based on the suggestion by @Huite in https://github.com/Deltares/Ribasim/issues/993#issuecomment-2187242512.
By repeatedly running `pixi run test-ribasim-qgis-ui` I am quite confident this fixes the issue. If I change the 0 back to 1 I got a crash again within a few tries.

Huite commented on ribasim_qgis/widgets/dataset_widget.py:
```diff
@@ -50,10 +50,10 @@ def __init__(self, parent: QWidget | None):
         self.setHeaderLabels([""])
         self.setHeaderHidden(False)
+        self.setColumnCount(1)
         header = self.header()
```
Do we still need any header logic? There's only a single column, so there's no sections to speak of. The resizing behavior only matters for multiple columns too.

Huite commented on ribasim_qgis/widgets/dataset_widget.py:
```diff
@@ -50,10 +50,10 @@ def __init__(self, parent: QWidget | None):
         self.setSizePolicy(QSizePolicy.Minimum, QSizePolicy.Preferred)
         self.setHeaderLabels([""])
```
Does it make sense to write `"  Layer"` here as a label here? (Note the spaces in front, otherwise it looks too cramped...)

![image](https://github.com/Deltares/Ribasim/assets/13662783/ec7cabeb-8c38-46b3-92ba-3305a196ce54)

Huite commented:
It makes sense that this fixes it (although still weird that doesn't just raises an exception).

I've approved, although take a look at my review comments -- since you're looking at this anyway.

visr commented:
Thanks, addressed comments in 7cd92b32776c2fae8d44cbc95f7b67850b47f2b5.

Looks good now:

![image](https://github.com/Deltares/Ribasim/assets/4471859/13274506-022a-473f-a926-b3dbad19b6c8)
